cpp/box/Box.h

```diff
@@ -380,6 +380,61 @@ class Box
         });
     }
 
+    //! Calculate distance between two points using boundary conditions
+    /*! \param r_i Position of first point
+        \param r_j Position of second point
+    */
+    inline float computeDistance(const vec3<float>& r_i, const vec3<float>& r_j) const
+    {
+            const vec3<float> r_ij = wrap(r_j - r_i);
+            return std::sqrt(dot(r_ij, r_ij));
+    }
+
+    //! Calculate distances between a set of query points and points.
+    /*! \param query_points Query point positions.
+        \param n_query_points The number of query points.
+        \param points Point positions.
+        \param n_points The number of points.
+        \param distances Distances between points and query_points (length is n_query_points).
+    */
+    void computeDistances(const vec3<float>* query_points, const unsigned int n_query_points,
+        const vec3<float>* points, const unsigned int n_points, float *distances) const
+    {
+        if (n_query_points != n_points)
+        {
+            throw std::invalid_argument("The number of query points and points must match.");
+        }
+        util::forLoopWrapper(0, n_query_points, [=](size_t begin, size_t end) {
```

Comment:
```suggestion
        util::forLoopWrapper(0, n_query_points, [&](size_t begin, size_t end) {
```

This will capture all the inputs by reference. That does make it the inner function's responsibility not to change the values but in this case that looks safe and saves some copying.